src/Service/DeliveryDateService.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/** @noinspection PhpUndefinedMethodInspection */
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Pdk\Service;
+
+use DateTime;
+use Exception;
+use MyParcelNL\Pdk\Base\Settings;
+use MyParcelNL\Pdk\Carrier\Model\DropOffDayPossibilities;
+use MyParcelNL\Pdk\Shipment\Collection\DropOffDayCollection;
+use MyParcelNL\Pdk\Shipment\Model\DropOffDay;
+
+class DeliveryDateService
+{
+    public const DATE_FORMAT = 'Y-m-d H:i:s';
+
+    /**
+     * @param  string|DateTime $deliveryDate
+     *
+     * @return string
+     * @throws \Exception
+     */
+    public static function fixPastDeliveryDate($deliveryDate): string
+    {
+        $tomorrow = new DateTime('tomorrow');
+
+        try {
+            $deliveryDateObject = is_a($deliveryDate, DateTime::class) ? $deliveryDate : new DateTime($deliveryDate);
+        } catch (Exception $e) {
+            return $tomorrow->format(self::DATE_FORMAT);
+        }
+
+        $oldDate = clone $deliveryDateObject;
+        $tomorrow->setTime(0, 0);
+        $oldDate->setTime(0, 0);
+
+        if ($deliveryDateObject < $tomorrow || '0' === $deliveryDateObject->format('w')) {
+            $deliveryDateObject = $tomorrow;
+        }
+
+        return $deliveryDateObject->format(self::DATE_FORMAT);
+    }
+
+    public static function getDeliveryDays(DateTime $today): DropOffDayCollection
+    {
+        $dropOffPossibilities = new DropOffDayPossibilities(Settings::get('settings'));
+        /** @var \MyParcelNL\Pdk\Shipment\Collection\DropOffDayCollection $dropOffDays */
+        $dropOffDays        = $dropOffPossibilities->getDropOffDays();
+        $exceptionDays      = self::getExceptionDays($dropOffPossibilities, $today);
+        $updatedDropOffDays = $dropOffDays->map(function (DropOffDay $dropOffDay) use ($exceptionDays) {
+            $exceptionDateCollection = $exceptionDays->where('date', '==', $dropOffDay->date);
```

Comment:
hier kan je firstWhere gebruiken, dan kan de hele collection shizzle weg en krijg je direct een DropOffDay in die method